src/main/java/roomescape/controller/ReservationController.java

```diff
@@ -2,45 +2,39 @@
 
 import java.net.URI;
 import java.util.List;
-import org.springframework.beans.factory.annotation.Autowired;
+
 import org.springframework.http.ResponseEntity;
-import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.DeleteMapping;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.ResponseBody;
-import roomescape.domain.Reservation;
-import roomescape.domain.Reservations;
-
-@Controller
+import org.springframework.web.bind.annotation.*;
+
+import roomescape.dto.ReservationRequestDTO;
+import roomescape.dto.ReservationResponseDTO;
+import roomescape.service.ReservationService;
+
+@RestController
 public class ReservationController {
 
-    @Autowired
-    private Reservations reservations;
+    private final ReservationService reservationService;
 
-    @GetMapping("/reservation")
-    public String reservation() {
-        return "reservation";
+    public ReservationController(ReservationService reservationService) {
+        this.reservationService = reservationService;
     }
 
     @GetMapping("/reservations")
-    @ResponseBody
-    public ResponseEntity<List<Reservation>> readReservations() {
-        return ResponseEntity.ok().body(reservations.getReservations());
+    public ResponseEntity<List<ReservationResponseDTO>> readReservations() {
+        List<ReservationResponseDTO> reservations = reservationService.findAll();
+        return ResponseEntity.ok(reservations);
```

Comment:
여기 부분은 고치는건 아니고 저도 몇시간전에 알게 됐는데요. 아래처럼 수정해도 200 OK에 body로 감싸서 나오더라구요 😲

이걸 알게 되면서 일관성 있는 코드를 작성하는게 규칙이라면 `ResponseEntity`를 넣는게 맞아보이지만, 진짜 필요한 코드만 적는게규칙이라면 200 OK에서는 Response Entity를 명시하지 않는게 맞지 않을까라는 생각이 들었습니다

```suggestion
    public List<ReservationResponseDTO> readReservations() {
        List<ReservationResponseDTO> reservations = reservationService.findAll();
        return reservations;
```

Reply:
```
package org.apache.coyote;

import //생략

public final class Response {
    private static final StringManager sm = StringManager.getManager(Response.class);
    private static final Log log = LogFactory.getLog(Response.class);
    private static final Locale DEFAULT_LOCALE = Locale.getDefault();
    int status = 200;
```
왜 그렇게 되는지 궁금증이 생겨서 찾아보았는데 Tomcat 내부의 coyote 패키지의 Response 클래스에 기본 상태를 200으로 지정해두고 있어 setStatus를 호출 하지 않으면 기본 상태가 200으로 설정 된다고 하더라고요! 궁금해서 직접 내부 코드를 찾아가 보았는데 실제로 그런 것 같았습니다. 200ok에서는 상황에 따라 ResponseEntity를 생략해도 괜찮겠네요!

Reply:
오 신기하네요? 상태코드는 무조건 줘야하는 데이터이니 정상 응답값을 반환한다면 기본적으로 200을 보내기로 한건가보군요